content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: Unreleased Production Changes Probe
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/blog_20230702.png
+sig:
+- gsoc
+---
+
+== Introduction
+
+A bug fix, security fix, new features are of no use unless they are released.
+
+A production release not only provides users with updated plugin but also give the developers a chance to get feedback on their work.
+
+The "unreleased production changes" probe looks for unreleased commits in the production files.
+
+If an unreleased file is found in the production path, the probe returns a failure message.
+If no unreleased file is found, it returns a success message.
+
+NOTE: for now, we are focusing on Maven production files (`pom.xml` and `src/main` folder). 
+
+This is my first probe that went live as a GSoC contributor.
+
+== Importance of the probe
+The probe aims to identify unreleased production commits in  Maven production files.
+
+Finding an unreleased production file can mean one or more of the following:
+
+* There might be unreleased security fixes that should be released as soon as possible.
+* There might be unreleased features requested by active users to improve the usefulness of the plugin.
+* One of the reasons for unreleased production changes could be an inactive or a super busy maintainer. This means the plugin needs a new maintainer, and it should be up for adoption.
```

Comment:
```suggestion
* The plugin maintainer may be inactive or too busy to further develop the plugin.
This can indicate the plugin needs a new maintainer, and it should be up for adoption.
```
Suggestion to rephrase to make this more inline with the two previous statements. If this is not the intended message, disregard!